src/Planner/Planner.cpp

```diff
@@ -1047,15 +1047,15 @@ PlannerContextPtr buildPlannerContext(const QueryTreeNodePtr & query_tree_node,
 }
 
 Planner::Planner(const QueryTreeNodePtr & query_tree_,
-    const SelectQueryOptions & select_query_options_)
+    SelectQueryOptions & select_query_options_)
```

Comment:
Why we started to pass this options by reference ?
It is totally unexpected that this can be changed outside of Planner.